pkg/writer/serializer_cdx14.go

```diff
@@ -182,43 +200,91 @@ func nodeToCDX14Component(n *sbom.Node) *cdx14.Component {
 	}
 
 	if n.Licenses != nil && len(n.Licenses) > 0 {
-		c.Licenses = []cdx14.License{}
+		var licenseChoices []cdx14.LicenseChoice
+		var licenses cdx14.Licenses
 		for _, l := range n.Licenses {
-			c.Licenses = append(c.Licenses, cdx14.License{
-				License: struct {
-					ID string "json:\"id\"" // TODO optimize
-				}{l},
+			licenseChoices = append(licenseChoices, cdx14.LicenseChoice{
+				License: &cdx14.License{
+					ID: l,
+				},
 			})
 		}
+
+		licenses = licenseChoices
+		c.Licenses = &licenses
 	}
 
 	if n.Hashes != nil && len(n.Hashes) > 0 {
-		c.Hashes = []cdx14.Hash{}
+		c.Hashes = &[]cdx14.Hash{}
 		for algo, hash := range n.Hashes {
-			c.Hashes = append(c.Hashes, cdx14.Hash{
-				Algorithm: algo, // Fix to make it valid
-				Content:   hash,
+			*c.Hashes = append(*c.Hashes, cdx14.Hash{
+				Algorithm: NormalizeAlgo(algo), // Fix to make it valid
+				Value:     hash,
 			})
 		}
 	}
 
 	if n.ExternalReferences != nil {
 		for _, er := range n.ExternalReferences {
 			if er.Type == "purl" {
-				c.Purl = er.Url
+				c.PackageURL = er.Url
 				continue
 			}
 
 			if c.ExternalReferences == nil {
-				c.ExternalReferences = []cdx14.ExternalReference{}
+				c.ExternalReferences = &[]cdx14.ExternalReference{}
 			}
 
-			c.ExternalReferences = append(c.ExternalReferences, cdx14.ExternalReference{
-				Type: er.Type,
+			*c.ExternalReferences = append(*c.ExternalReferences, cdx14.ExternalReference{
+				Type: cdx14.ExternalReferenceType(er.Type), // Fix to make it valid
 				URL:  er.Url,
 			})
 		}
 	}
 
 	return c
 }
+
+func NormalizeAlgo(algo string) cdx14.HashAlgorithm {
```

Comment:
I'm going to merge this function but I think we should overload the proto with methods to output to each format. I'll do it in a follow-up to iterate faster, I'll tag you so that you can tell me what you think.